scalyr_agent/log_processing.py

```diff
@@ -2162,7 +2162,7 @@ def __create_events_object(self, line_object, sampling_rate):
         """
         result = Event(base=self.__base_event)
         if line_object.attrs:
-            result.add_missing_attributes(line_object.attrs)
+            result.add_missing_attributes(line_object.attrs, log_line_attributes=True)
```

Comment:
Hmm.. I don't know why we only set the attributes if the event doesn't already have a value for the attributes.  If the base event has a value for `stream`, we want the attribute for this event to override it.  Meaning, if the log attributes have a value of `stream=foo` and the line has an attribute of `stream=bar`, we want `bar` to win.  if you look at the implementation of `add_missing_attribute`, we'll only update the attributes if `stream` is not present.

Reply:
For cases where the log line attributes how does the server interpret both the log having the attribute and the event? Which wins out, and does the server even read event attributes if it has log attributes?

If log ones win out anyway we may not have a good way to do things on our end to make the event ones win out.

Reply:
For now I hid overwriting behind `log_line_attributes=True` in case somewhere in the depths the code relies on it not being overwritten, though I couldn't even begin to guess where or why.